mod-dcb/src/main/resources/volaris/mod-dcb/features/lending-flow.feature

```diff
@@ -0,0 +1,509 @@
+    "returnedLostItemProcessingFee": true,
+    "replacedLostItemProcessingFee": true,
+    "replacementProcessingFee": 0.0,
+    "replacementAllowed": true,
+    "lostItemReturned": "Charge",
+    "id": "ed892c0e-52e0-4cd9-8133-c0ef07b4a709"
+    }
+    """
+    When method POST
+
+  @CirculationRules
+  Scenario: Update circulation rules
+    Given path 'circulation/rules'
+    And request
+    """
+    {
+    "id": "1721f01b-e69d-5c4c-5df2-523428a04c55",
+    "rulesAsText": "priority: t, s, c, b, a, m, g\nfallback-policy: l d9cd0bed-1b49-4b5e-a7bd-064b8d177231 r d9cd0bed-1b49-4b5e-a7bd-064b8d177231 n 122b3d2b-4788-4f1e-9117-56daa91cb75c o cd3f6cac-fa17-4079-9fae-2fb28e521412 i ed892c0e-52e0-4cd9-8133-c0ef07b4a709 \nm 1a54b431-2e4f-452d-9cae-9cee66c9a892: l d9cd0bed-1b49-4b5e-a7bd-064b8d177231 r d9cd0bed-1b49-4b5e-a7bd-064b8d177231 n 122b3d2b-4788-4f1e-9117-56daa91cb75c o cd3f6cac-fa17-4079-9fae-2fb28e521412 i ed892c0e-52e0-4cd9-8133-c0ef07b4a709"
+    }
+    """
+    When method PUT
+
+  Scenario: Create DCB Transaction
+    * def transaction = read('classpath:volaris/mod-dcb/features/samples/transaction/create-dcb-transaction.json')
+
+    Given path 'transactions' , dcbTransactionId
+    And request transaction
+    When method POST
+    Then status 201
+    And match $.status == 'CREATED'
+    And match $.item.id == extItemId
+    And match $.patron.id == patronId
+
+  Scenario: Get Item status after creating dcb transaction
+
+    Given path 'item-storage', 'items', extItemId
+    When method GET
+    Then status 200
+    And match $.barcode == itemBarcode
+    And match $.status.name == 'Paged'
+
+
+  Scenario: Get User Type  after creating dcb transaction
+
+    Given path '/users/' + patronId
+    When method GET
+    Then status 200
+    And match $.barcode == patronBarcode
+    And match $.type == 'dcb'
+
+  Scenario: Get request by barcode and item ID after creating dcb transaction
+
+    Given path 'request-storage', 'requests'
+    Given param query = '(item.barcode= ' + itemBarcode + ' and itemId = ' + extItemId + ' )'
+    When method GET
+    Then status 200
+    And match $.totalRecords == 1
+    And match $.requests[0].status == 'Open - Not yet filled'
+
+  Scenario: Get loan by item ID after creating dcb transaction
+
+    Given path 'loan-storage', 'loans'
+    Given param query = '( itemId = ' + extItemId + ' )'
+    When method GET
+    Then status 200
+    And match $.totalRecords == 0
+
+  Scenario: Check Transaction status after creating dcb transaction
+
+    Given path 'transactions' , dcbTransactionId , 'status'
+    When method GET
+    Then status 200
+    And match $.status == 'CREATED'
+    And match $.role == 'LENDER'
+
+  Scenario: current item check-in record and its status
+    * def intCheckInDate = call read('classpath:volaris/mod-dcb/features/util/get-time-now-function.js')
+    * def checkInRequest = read('classpath:volaris/mod-dcb/features/samples/check-in/check-in-by-barcode-entity-request.json')
+
```

Comment:
is it possible to move all prerequisite resources to another feature file . may be in a separate folder named reusable.

why because these would be needed for all DCB roles.

Reply:
Added